drivers/gps/nrf9160_gps/nrf9160_gps.c

```diff
@@ -389,58 +305,54 @@ static void gps_thread(int dev_ptr)
 				 * duration and GPS priority is set by the
 				 * application, GPS priority is requested.
 				 */
-				if (drv_data->current_cfg.priority) {
+				if (drv_data->config.priority) {
 					k_work_schedule(
 						&drv_data->blocked_work,
 						K_SECONDS(GPS_BLOCKED_TIMEOUT));
 				}
 
-				continue;
-			} else if (operation_blocked) {
+				goto free_event;
+			} else if (atomic_get(&drv_data->operation_blocked)) {
 				/* GPS has been unblocked. */
 				LOG_DBG("GPS has time window to operate");
 
-				operation_blocked = false;
+				atomic_set(&drv_data->operation_blocked, 0);
 				evt.type = GPS_EVT_OPERATION_UNBLOCKED;
 
 				notify_event(dev, &evt);
 
-				/* This may fail when the work has already
-				 * started, and the GPS priority will be
-				 * then requested anyway.
-				 */
-				k_work_cancel_delayable_sync(
-					&drv_data->blocked_work,
-					&drv_data->work_sync);
+				k_work_cancel_delayable(&drv_data->blocked_work);
```

Comment:
Would be better to call k_work_cancel_delayable_sync() instead as it was done earlier. This is a executed on the context of the GPS thread so no need to worry about stalling an event queue.

Reply:
Same question here.